tests/unit/utils.py

```diff
@@ -45,3 +45,48 @@ def tearDown(self):
     @property
     def last_client_output(self):
         return self.client_output[-1] if self.client_output else None
+
+class DustyIntegrationTestCase(TestCase):
+    """This test case intentionally avoids mocking whenever possible
+    in order to get as close as possible to the actual state that
+    would be experienced on a system running Dusty. Therefore,
+    integration tests are possibly destructive if run on a user's
+    machine. To help protect the user from running them accidentally,
+    integration tests will refuse to run unless the environment
+    variable DUSTY_ALLOW_INTEGRATION_TESTS is set.
+
+    Note that this also assumes it is running against an actual
+    Dustyd process on the local host."""
+    def setUp(self):
+        if not os.getenv('DUSTY_ALLOW_INTEGRATION_TESTS'):
+            raise RuntimeError('You must set the env var DUSTY_ALLOW_INTEGRATION_TESTS to run integration tests. '
+                               'This may affect your local config, do not run integration tests on your actual '
+                               "machine unless you know what you're doing!")
+        self.overridden_specs_path = tempfile.mkdtemp()
+        write_default_config()
+        save_config_value('specs_repo', 'github.com/gamechanger/example-dusty-specs')
+        override_repo(get_specs_repo(), self.overridden_specs_path)
+
+    def tearDown(self):
+        shutil.rmtree(self.overridden_specs_path)
+
+    @property
+    def stdout(self):
+        return sys.stdout.getvalue().strip()
```

Comment:
ah dammit in my original PR i moved this entire file, ok so that's why everything went to shit. yeah this guy needs to not live in `unit` anymore. i'll move it in a separate PR once this is merged.